src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -1840,19 +1932,51 @@ class NodeUtils(object):
         host_resc_avail = vnode_list[vnode_name].resources_available
         pbs.logmsg(pbs.EVENT_DEBUG4, '%s: host_resc_avail: %s' %
                    (caller_name(), host_resc_avail))
-        # Set the vnode type if supplied
-        if vntype:
+
+        # Set resources_available.vntype of natural node according
+        # to what's in local file if it needs to be propagated to server
+        if (vntype and "propagate_vntype_to_server" in self.cfg
+                and self.cfg['propagate_vntype_to_server']):
```

Comment:
We should never have to check whether a configuration value exists. Please set a default value in parse_config_file().

Reply:
A default is already set:
        defaults['propagate_vntype_to_server'] = True
so you are correct this is redundant.